diff --git a/litert/CMakeLists.txt b/litert/CMakeLists.txt
index aaa78dc9..5709f9e9 100644
--- a/litert/CMakeLists.txt
+++ b/litert/CMakeLists.txt
@@ -15,6 +15,10 @@
 cmake_minimum_required(VERSION 3.20)
 project(LiteRT VERSION 1.4.0 LANGUAGES CXX C)
 
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive" CACHE STRING "Required CXX flags" FORCE)
+
+
 function(_litert_overlay_converter_sources TF_SRC_ROOT)
   set(_overlay_root "${CMAKE_CURRENT_SOURCE_DIR}/../tflite/converter")
   if(NOT EXISTS "${TF_SRC_ROOT}")
@@ -63,17 +67,17 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-if(CMAKE_SYSTEM_PROCESSOR)
-  string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" LITERT_SYSTEM_PROCESSOR_LOWER)
-else()
-  set(LITERT_SYSTEM_PROCESSOR_LOWER "")
-endif()
-set(_litert_disable_kleidiai_default OFF)
-if(LITERT_SYSTEM_PROCESSOR_LOWER MATCHES "^(x86_64|amd64|i[3-6]86)$")
-  if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    set(_litert_disable_kleidiai_default ON)
-  endif()
-endif()
+# if(CMAKE_SYSTEM_PROCESSOR)
+#   string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" LITERT_SYSTEM_PROCESSOR_LOWER)
+# else()
+#   set(LITERT_SYSTEM_PROCESSOR_LOWER "")
+# endif()
+# set(_litert_disable_kleidiai_default OFF)
+# if(LITERT_SYSTEM_PROCESSOR_LOWER MATCHES "^(x86_64|amd64|i[3-6]86)$")
+#   if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+#     set(_litert_disable_kleidiai_default ON)
+#   endif()
+# endif()
 
 # Options
 option(LITERT_BUILD_SHARED_LIBS "Build shared libraries" OFF)
diff --git a/litert/c/CMakeLists.txt b/litert/c/CMakeLists.txt
index 14df315b..e971da7c 100644
--- a/litert/c/CMakeLists.txt
+++ b/litert/c/CMakeLists.txt
@@ -180,7 +180,7 @@ set(_litert_runtime_c_api_private_archives
     "${TFLITE_BUILD_DIR}/libxnnpack-delegate.a"
     "${TFLITE_BUILD_DIR}/_deps/xnnpack-build/libXNNPACK.a"
     "${TFLITE_BUILD_DIR}/_deps/xnnpack-build/libxnnpack-microkernels-prod.a"
-    "${TFLITE_BUILD_DIR}/kleidiai/libkleidiai.a"
+    #"${TFLITE_BUILD_DIR}/kleidiai/libkleidiai.a"
     "${TFLITE_BUILD_DIR}/_deps/ruy-build/ruy/libruy_allocator.a"
     "${TFLITE_BUILD_DIR}/_deps/ruy-build/ruy/libruy_apply_multiplier.a"
     "${TFLITE_BUILD_DIR}/_deps/ruy-build/ruy/libruy_block_map.a"
diff --git a/litert/compiler/CMakeLists.txt b/litert/compiler/CMakeLists.txt
index d39485eb..7a94cb52 100644
--- a/litert/compiler/CMakeLists.txt
+++ b/litert/compiler/CMakeLists.txt
@@ -36,6 +36,7 @@ target_link_libraries(litert_compiler_plugin
         litert_core
         litert_core_model
         litert_c_api
+        litert_runtime
         absl::status
         absl::statusor
         absl::strings
diff --git a/litert/runtime/compiled_model.cc b/litert/runtime/compiled_model.cc
index 10968348..fef81bb0 100644
--- a/litert/runtime/compiled_model.cc
+++ b/litert/runtime/compiled_model.cc
@@ -801,7 +801,8 @@ LiteRtCompiledModelT::GetTensorBufferRequirements(const TfLiteTensor* tensor) {
       /*num_strides=*/1, cpu_buffer_strides, &litert_cpu_buffer_requirements));
   cpu_buffer_requirements_[tensor_id] =
       LiteRtTensorBufferRequirementsPtr(litert_cpu_buffer_requirements);
-  return litert_cpu_buffer_requirements;
+  return litert::Expected<const LiteRtTensorBufferRequirementsT*>(
+    litert_cpu_buffer_requirements);
 }
 
 Expected<const LiteRtTensorBufferRequirementsT*>
diff --git a/litert/tools/CMakeLists.txt b/litert/tools/CMakeLists.txt
index 88bbdae0..f824f16c 100644
--- a/litert/tools/CMakeLists.txt
+++ b/litert/tools/CMakeLists.txt
@@ -163,11 +163,21 @@ target_include_directories(analyze_model
 
 target_link_libraries(analyze_model
     PRIVATE
+            
+        litert_c_api
+        litert_cc_api
+        
         litert_dump
         litert_tool_display
         litert_core
         litert_core_model
+
+        litert_runtime
+        litert_logging
+        absl::log_internal_check_op
+
         tensorflow-lite
+        absl::base
         absl::flags
         absl::flags_parse
         absl::log
@@ -189,6 +199,9 @@ target_include_directories(apply_plugin_main
 
 target_link_libraries(apply_plugin_main
     PRIVATE
+        litert_c_api
+        litert_cc_api
+        $<LINK_LIBRARY:WHOLE_ARCHIVE,litert_runtime>
         litert_apply_plugin
         litert_dump
         litert_tool_flags_apply_plugin
@@ -197,12 +210,11 @@ target_link_libraries(apply_plugin_main
         litert_tool_flags_intel_openvino
         litert_tool_flags_mediatek
         litert_tool_flags_qualcomm
-
         litert_tool_display
         litert_core
         litert_core_model
-        litert_runtime
         tensorflow-lite
+        absl::base
         absl::flags
         absl::flags_parse
         absl::log
